Formula/massren.rb

```diff
@@ -0,0 +1,55 @@
+require "language/go"
+
+class Massren < Formula
+  desc "Easily rename multiple files using your text editor"
+  homepage "https://github.com/laurent22/massren"
+  url "https://github.com/laurent22/massren/archive/v1.5.1.tar.gz"
+  sha256 "21a602a29410e30b1b518356d5707b25b22b9d2578aaf0d0d5ab3de9e0ad225d"
+
+  depends_on "go" => :build
+
+  go_resource "golang.org/x/net" do
+    url "https://go.googlesource.com/net.git",
+        :revsion => "3f122ce3dbbe488b7e6a8bdb26f41edec852a40b"
+  end
+
+  go_resource "github.com/jessevdk/go-flags" do
+    url "https://github.com/jessevdk/go-flags.git",
+        :revision => "f88afde2fa19a30cf50ba4b05b3d13bc6bae3079"
+  end
+
+  go_resource "github.com/kr/text" do
+    url "https://github.com/kr/text.git",
+        :revision => "7cafcd837844e784b526369c9bce262804aebc60"
+  end
+
+  go_resource "github.com/laurent22/go-sqlkv" do
+    url "https://github.com/laurent22/go-sqlkv.git",
+        :revision => "db1022af2c66c36268b346d203ceaa78579673d2"
+  end
+
+  go_resource "github.com/laurent22/go-trash" do
+    url "https://github.com/laurent22/go-trash.git",
+        :revision => "681610e5e1bc5214826533dfa339c84e71e49821"
+  end
+
+  go_resource "github.com/mattn/go-sqlite3" do
+    url "https://github.com/mattn/go-sqlite3.git",
+        :revision => "ca5e3819723d8eeaf170ad510e7da1d6d2e94a08"
+  end
+
+  go_resource "github.com/nu7hatch/gouuid" do
+    url "https://github.com/nu7hatch/gouuid.git",
+        :revision => "179d4d0c4d8d407a32af483c2354df1d2c91e6c3"
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath
+    Language::Go.stage_deps resources, buildpath/"src"
+    system "go", "build", "-o", bin/"massren"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/massren --version")
```

Comment:
This needs to be a test that exercises and verifies the functionality of the software.

Reply:
That’s not easily feasible. What `massren` does is grab all the filenames from the current directory and open them in a text editor. Then after we save and close the text file, it renames them.

That is to say it’s a tool that only works interactively; it’s not for automation.

I guess we could try make it work by temporarily changing `EDITOR` to some `vim` call that makes some simple text changes, but `vim` is so customisable, I’m wary of doing that and messing something up in the user’s machine.

Reply:
The test block runs in a macOS sandbox and in a temporary directory so messing something up on the user's machine isn't really possible. The `vim` formula's test block may help shed some light: https://github.com/Homebrew/homebrew-core/blob/master/Formula/vim.rb#L123-L143

Reply:
I’m trying to do

```ruby
  system bin/"massren", "--config", "editor", "/usr/bin/nano"
  assert_match 'editor = "nano"', shell_output("#{bin}/massren --config")
```

But when running `brew test massren` locally, I get:

```ruby
/usr/local/Cellar/massren/1.5.1/bin/massren
--config
editor
/usr/bin/nano

panic: attempt to write a readonly database
```

Guessing the database is readonly because we’re in the sandbox. Trouble is I then try some more convoluted method to actually do some renaming instead, but since `massren` (seems that it) uses a temporary file in `~/.config/massren` to keep track of the renaming, I get a `stat` error saying the temporary file does not exist (again I suppose it’s due to the sandbox).

Reply:
During the test, the environment variable HOME is set equal to the `testpath`. So for example
```
bash-3.2$ echo $HOME
/tmp/wget-test-20171024-15182-1g3a9jz
bash-3.2$ 
```
It sounds like the software isn't respecting the HOME environment variable.

Reply:
Yeah, https://github.com/laurent22/massren/blob/4de39fa7117e4a8a40805ec17d8d584aa5253452/profile.go#L71-L93 disregards HOME.

Reply:
We ran into a similar issue the other days which was fixed here: https://github.com/ssh-vault/ssh-vault/commit/33b59a33979223c2a5e7e5b63f0bd01e50fc04ae

Reply:
That's weird, it's using Go's `user.Current().HomeDir` which I would have assumed is the same as HOME (at least it seems to be on most systems) but I guess not in this case. Thanks for link, it can indeed be fixed using `os.Getenv("HOME")`.

Reply:
Thanks again @laurent22 :)